plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -164,7 +165,7 @@ public CheckpointEntryIterator(
         if (fields.contains(ADD)) {
             checkArgument(metadataEntry.isPresent(), "Metadata entry must be provided when reading ADD entries from Checkpoint files");
             this.metadataEntry = metadataEntry.get();
-            this.schema = extractSchema(this.metadataEntry, typeManager);
+            this.schema = extractSchema(this.metadataEntry, typeManager).stream().map(DeltaLakeColumnMetadata::getColumnMetadata).collect(toImmutableList());
```

Comment:
I see this logic duplicated also in `TransactionLogAccess` and `CheckpointManager`.
Maybe it is worth exposing a method in  `DeltaLakeSchemaSupport` for avoiding code duplication.